pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -92,23 +92,17 @@ func (c *defaultControl) deleteShoot(o *operation.Operation) *gardenv1beta1.Last
 	// We also check whether the kube-apiserver pod exists in the Shoot namespace within the Seed cluster. If it does not,
 	// then we assume that it has never been deployed successfully. We follow that no resources can have been deployed
 	// at all in the Shoot cluster, thus there is nothing to delete at all.
-	kubeAPIServerFound := false
-	podList, err := botanist.K8sSeedClient.ListPods(o.Shoot.SeedNamespace, metav1.ListOptions{
-		LabelSelector: "app=kubernetes,role=apiserver",
-	})
-	if err != nil {
-		return formatError("Failed to retrieve the list of pods running in the Shoot namespace in the Seed cluster", err)
-	}
-	for _, pod := range podList.Items {
-		if pod.DeletionTimestamp == nil {
-			kubeAPIServerFound = true
-			break
+	noDeletionCheckpointFound := false
```

Comment:
Instead of using a `ConfigMap` with a 'magic name' why don't we just use the kube API server deployment? This should still be present but just be scaled to 0.

Reply:
The kube-apiserver deployment is deleted as part of the shoot reconciliation flow

Reply:
The API server is just scaled down, the deployment should still exist (with `0` replicas), shouldn't it? If yes, then we can make the assumption that we need to clean up as long as there is / was such a deployment.

Reply:
> [The kube-apiserver deployment is deleted as part of the shoot reconciliation flow](https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/shoot/shoot_control_delete.go#L245-L249)

If we would only scale it down we cannot differentiate whether we have to cleanup the Kubernetes resources or not (users can delete shoots that are hibernated already and here the resp. kube-apiserver is scaled down already).